Polyfills/WebSocket/Source/WebSocket.cpp

```diff
@@ -0,0 +1,244 @@
+#include "WebSocket.h"
+#include <Babylon/JsRuntime.h>
+
+namespace Babylon::Polyfills::Internal
+{
+    void WebSocket::Initialize(Napi::Env env)
+    {
+        static constexpr auto JS_WEB_SOCKET_CONSTRUCTOR_NAME = "WebSocket";
+
+        if (env.Global().Get(JS_WEB_SOCKET_CONSTRUCTOR_NAME).IsUndefined())
+        {
+            Napi::Function func = DefineClass(
+                env,
+                JS_WEB_SOCKET_CONSTRUCTOR_NAME,
+                {
+                    StaticValue("CONNECTING", Napi::Value::From(env, 0)),
+                    StaticValue("OPEN", Napi::Value::From(env, 1)),
+                    StaticValue("CLOSING", Napi::Value::From(env, 2)),
+                    StaticValue("CLOSED", Napi::Value::From(env, 3)),
+                    InstanceAccessor("readyState", &WebSocket::GetReadyState, nullptr),
+                    InstanceAccessor("url", &WebSocket::GetURL, nullptr),
+                    InstanceAccessor("onopen", &WebSocket::GetOnOpen, &WebSocket::SetOnOpen),
+                    InstanceAccessor("onclose", &WebSocket::GetOnClose, &WebSocket::SetOnClose),
+                    InstanceAccessor("onmessage", &WebSocket::GetOnMessage, &WebSocket::SetOnMessage),
+                    InstanceAccessor("onerror", &WebSocket::GetOnError, &WebSocket::SetOnError),
+                    InstanceMethod("close", &WebSocket::Close),
+                    InstanceMethod("send", &WebSocket::Send),
+                });
+
+            env.Global().Set(JS_WEB_SOCKET_CONSTRUCTOR_NAME, func);
+        }
+    }
+
+    WebSocket::WebSocket(const Napi::CallbackInfo& info)
+        : Napi::ObjectWrap<WebSocket>{info}
+        , m_runtimeScheduler{JsRuntime::GetFromJavaScript(info.Env())}
+        , m_webSocket(
+              info[0].As<Napi::String>(),
+              [this, env = info.Env()]() {
```

Comment:
nit: Maybe put this into individual member functions so that you can put each callback on one line to make this more readable.